code/go/0chain.net/smartcontract/dbs/event/snapshot.go

```diff
@@ -0,0 +1,59 @@
+package event
+
+import (
+	"errors"
+	"sync"
+
+	"0chain.net/core/logging"
+	"go.uber.org/zap"
+	"gorm.io/gorm"
+)
+
+// swagger:model Block
+type Snapshot struct {
+	gorm.Model
+	Round           int64  `gorm:"primaryKey;autoIncrement:false" json:"round"`
+	BlockHash       string `json:"block_hash"`
+	MintTotalAmount int64  `json:"mint_total_amount"`
+}
+
+var snapshotLock sync.Mutex
+
+func (edb *EventDb) GetRoundsMintTotal(from, to int64) (int64, error) {
+	var total int64
+	return total, edb.Store.Get().Model(&Snapshot{}).Where("round between ? and ?", from, to).Select("sum(mint_total_amount)").Scan(&total).Error
+}
+
+func (edb *EventDb) addOrUpdateTotalMint(mint Mint) error {
+	snapshotLock.Lock()
```

Comment:
the problem with this solution is that we can't protect data from update from other function that does not share this lock, even that we handle data change from one place in go code it is buggy and dev can forget or doesn't know about this lock. The best practice is to use SELECT ... FOR UPDATE construction. look for examplehere https://stackoverflow.com/questions/18879584/postgres-select-for-update-in-functions